code/__HELPERS/text.dm

```diff
@@ -904,6 +904,13 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 	if(abs(value) <= 1e-18)
 		return "0 [unit]"
 
+
+	// The game code never uses Pa, but kPa, since 1 Pa is too small to reasonably handle
+	// thus, to ensure correct conversion from any kPa in game code, this value needs to be multiplied by 10e3 to get Pa
+	if(unit == "Pa")
+		value = value * 1000
+
```

Comment:
Does this match the other bits of code that use Pa here? I have a feeling this would break some assumptions.

Reply:
This is a very ugly hack, make a new proc instead specifically made for si-printing pressure.

Reply:
> Does this match the other bits of code that use Pa here? I have a feeling this would break some assumptions.

Which assumptions do you mean? As far as i can tell, if this proc is used it always uses "Pa" to reference pressure.

Reply:
> This is a very ugly hack, make a new proc instead specifically made for si-printing pressure.

I feel like this is a little much for such a small proc, but the only way I think would make sense is to implement this for most of the standard SIs without tossing the previous coders work out the window (because their solution is rather nice). is basically a proc that is 

`proc/siunit_pressure(value_in_kpa, maxdecimals)`
`var/pressure_adj = value_in_kpa * 1000 //to adjust for using kPa instead of Pa`
`return siunit(pressure_adj, "Pa", maxdecimals)`

And then that for all other units that use the same proc in code (which isnt that much). if feels even more cumbersome. Or what did you have in mind?

Reply:
and all the other SI dont really need this adjustment and can just call siunit(). Its really just kPa that is the troublemaker.

Reply:
@Donkie Im toying around with your idea and tried the approach i detailed above. Should I implement it? so far it seems to work, and it does not break other SIs.